File: backend/flask/app/middleware/jwt_middleware.py
from flask import request, jsonify
from app.services.jwt_service import JwtService

def jwt_middleware(func):
    def wrapper(*args, **kwargs):
        token = request.headers.get('Authorization')
        if not token or not JwtService.validate_token(token):
            return jsonify({'error': 'Unauthorized'}), 401
        return func(*args, **kwargs)
    return wrapper

File: backend/laravel/app/Http/Middleware/JwtMiddleware.php
<?php

namespace App\Http\Middleware;

use App\Services\JwtService;
use Closure;
use Illuminate\Http\Request;

class JwtMiddleware
{
    private $jwtService;

    public function __construct(JwtService $jwtService)
    {
        $this->jwtService = $jwtService;
    }

    public function handle(Request $request, Closure $next)
    {
        $token = $request->header('Authorization');

        if (!$token || !$this->jwtService->validateToken($token)) {
            return response()->json(['error' => 'Unauthorized'], 401);
        }

        return $next($request);
    }
}
